### ISSUE TITLE
Potential issue piping binary file 

### ISSUE THREAD

**Issue by foolnotion** [sharkdp/bat#816]
Hi,

I was running some tests for a random number generator using the methodology described here: [how to test with PractRand](http://www.pcg-random.org/posts/how-to-test-with-practrand.html).

So I generate a ~3.8 gb file full of random bytes in packed binary format. Then the issue: `bat` produces different behavior compared to `cat`:

```
$ bat fastrand.dat | ./RNG_test stdin
bashRNG_test using PractRand version 0.95
RNG = RNG_stdin, seed = unknown
test set = core, folding = standard(unknown format)
rng=RNG_stdin, seed=unknown
length= 64 megabytes (2^26 bytes), time= 2.7 seconds
  Test Name                         Raw       Processed     Evaluation
  BCFN(2+0,13-3,T)                  R>+99999  p = 0           FAIL !!!!!!!!  
  BCFN(2+1,13-3,T)                  R>+99999  p = 0           FAIL !!!!!!!!  
  BCFN(2+2,13-4,T)                  R>+99999  p = 0           FAIL !!!!!!!!  
...etc...
...and 134 test result(s) without anomalies
```

I tried also with options `-A`, `-p`, `-pp` but they did not change anything. 
By comparison, `cat` produces:
```bash
$ cat fastrand.dat | ./RNG_test stdin
RNG_test using PractRand version 0.95
RNG = RNG_stdin, seed = unknown
test set = core, folding = standard(unknown format)
rng=RNG_stdin, seed=unknown
length= 128 megabytes (2^27 bytes), time= 2.2 seconds
  Test Name                         Raw       Processed     Evaluation
  BCFN(2+0,13-3,T)                  R>+99999  p = 0           FAIL !!!!!!!!  
  BCFN(2+1,13-3,T)                  R>+99999  p = 0           FAIL !!!!!!!!  
  BCFN(2+2,13-3,T)                  R>+99999  p = 0           FAIL !!!!!!!!  
...etc...
...and 148 test result(s) without anomalies
```

Notice the difference in the lines `length=64 megabytes` vs `length=128 megabytes` between `bat` and `cat`, as well as the different results obtained. 

Unless I am missing some CLI switch, I think this might be a bug in the way `bat` pipes binary input.

**Comment by sharkdp** [comment]
Thank you for reporting this.

Could you please pipe into `md5sum` to check if you really get a different output?

Which version of `bat` are you using?

**Comment by foolnotion** [comment]
I am using bat 0.12.1, and `md5sum` returns the same output for both (on a side note, it takes `cat` 7.4 seconds to pipe this file to `md5sum`, while `bat` takes 106 seconds --  I will test this more thoroughly when I am back to my desktop machine as this laptop is old). So I'm guessing it's not really a bug, but something about `PractRand`.

**Comment by foolnotion** [comment]
I can reproduce the same issue with PractRand on my desktop machine (NixOS, Ryzen 3900X, bat 0.12.1). Although the `md5sum` is the same, `PractRand` seems to be sensitive to the speed of the pipe. 

To test this hypothesis I used the [pv](https://linux.die.net/man/1/pv) utility to measure pipe speed:

- bat 37.9 MiB/s
- cat 55.0 MiB/s

I then used `pv` to limit the rate through the pipe and this is leading to exactly the same `PractRand` output for both `cat` and `bat`.

`bat fastrand.dat | pc -L 33m | ./RNG_test stdin`
